lib/BloqadeCUDA/test/subspace.jl

```diff
@@ -0,0 +1,65 @@
+using Test
+using Random
+using CUDA
+using YaoSubspaceArrayReg
+using YaoArrayRegister
+using StatsBase
+using CuYao
+using GPUArrays
+using Adapt
+using BloqadeCUDA
+
+
+struct Sampler{A,B}
```

Comment:
maybe we can start with a simple version without these abstractions

Reply:
Sure, I can't seem to figure out how to make the kernel compile without the abstractions. 

It is complaining about `searchsortedfirst` being called inside the map kernel.